### DISCUSSION TITLE
Unable to uncomment Golang struct

### DISCUSSION THREAD

**Discussion by smartinellimarco** [folke/ts-comments.nvim#36] [Q&A]
Im unable to uncomment a piece of code like:

```go
// type Node struct {
// 	NodeParams
// 	ID string `json:"id"`
// }
```
using the following settings:

```lua
local M = { 'folke/ts-comments.nvim' }

M.event = { 'VeryLazy' }
M.opts = {
  lang = {
    lua = '-- %s',
    go = '// %s',
  },
}

function M.config(_, opts)
  require('ts-comments').setup(opts)
end

return M

-- vim: ts=2 sts=2 sw=2 et
```
When I check the :messsages I get "Option 'commentstring' is empty."

**Comment by folke** [comment]
Should probably be golang?

**Reply by smartinellimarco** [reply]
I've already tried
Commenting works fine, just uncommenting doesn't

**Reply by smartinellimarco** [reply]
@folke I think its a more general issue. I'm seeing the same thing happen, depending on the treesitter node type I'm performing the comment/uncomment, for languages like Lua and Python.
I'm not using the builtin ftplugins by the way

**Reply by folke** [reply]
If you're not using built-in ftplugins then that's obviously your issue. 

Ts-comments simply configures neovims built-in comments. That uses ftdetect...

**Reply by smartinellimarco** [reply]
@folke I do have ft detection on, only turned off the script that defines the default commentstring (among other things), which I dont mind setting by hand with this plugin.
Is there a chance to use the one passsed to the options as the default if no commenstring was defined previously for that buffer?

**Reply by folke** [reply]
Sorry, but then you'r on your own. Internally Neovim's comments relies on corectly setup treesitter lang mappings and correctly working filetypes....
Commenting and uncommenting in go files works perfectly fine for me.

The fact that you chose not to tell from the start that you disabled ftplugins is also not very nice. 
You probably knew very well that this is the reason it did not work for you.
That's just wasting my time tbh. Not nice.

**Comment by smartinellimarco** [comment]
> The fact that you chose not to tell from the start that you disabled ftplugins is also not very nice.

I originally disabled ftplugins to prevent 'formatopts' from being overwritten, based on what I found while researching online. It took me some time to find the actual plugin scripts and discover they set a commentstring value, which is why I clarified it in a comment and asked if I could just set it using this plugin.

> You probably knew very well that this is the reason it did not work for you.
That's just wasting my time tbh. Not nice.

I don't appreciate the tone. I didn't know that was the root cause of the issue.

Anyways, thanks for your help.
